Sources/BaggageContext/Context.swift

```diff
@@ -0,0 +1,220 @@
+///
+/// ### Logged Metadata and Baggage Items
+///
+/// Please refer to your configured log handler documentation about how to configure which metadata values should be logged
+/// and which not, as each log handler may handle and configure those differently. The default implementations log *all*
+/// metadata/baggage values present, which often is the right thing, however in larger systems one may want to choose a
+/// log handler which allows for configuring these details.
+///
+/// ### Accepting context types in APIs
+///
+/// It is preferred to accept values of `ContextProtocol` in library APIs, as this yields a more flexible API shape,
+/// to which other libraries/frameworks may pass their specific context objects.
+///
+/// - SeeAlso: `Baggage` from the Baggage module.
+/// - SeeAlso: `Logger` from the SwiftLog package.
+public struct DefaultContext: Context {
+    /// The `Baggage` carried with this context.
+    /// It's values will automatically be made available to the `logger` as metadata when logging.
+    ///
+    /// Baggage values are different from plain logging metadata in that they are intended to be
+    /// carried across process and node boundaries (serialized and deserialized) and are made
+    /// available to instruments using `swift-distributed-tracing`.
+    public var baggage: Baggage
+
+    public var logger: Logger {
+        get {
+            return self._logger.with(self.baggage)
+        }
+        set {
+            self._logger = newValue
+        }
+    }
+
+    private var _logger: Logger
+
+    public init(baggage: Baggage, logger underlying: Logger) {
+        self.baggage = baggage
+        self._logger = underlying
+    }
+
+    public init<C>(context: C) where C: Context {
+        self._logger = context.logger
+        self.baggage = context.baggage
+    }
+}
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: `with...` functions
+
+extension DefaultContext {
+    /// Fluent API allowing for modification of underlying logger when passing the context to other functions.
+    ///
+    /// - Parameter logger: Logger that should replace the underlying logger of this context.
+    /// - Returns: new context, with the passed in `logger`
+    public func withLogger(_ function: (inout Logger) -> Void) -> DefaultContext {
+        var logger = self._logger
+        function(&logger)
+        return .init(baggage: self.baggage, logger: logger)
+    }
+
+    /// Fluent API allowing for modification of underlying log level when passing the context to other functions.
+    ///
+    /// - Parameter logLevel: New log level which should be used to create the new context
+    /// - Returns: new context, with the passed in `logLevel` used for the underlying logger
+    public func withLogLevel(_ logLevel: Logger.Level) -> DefaultContext {
+        var copy = self
+        copy.logger.logLevel = logLevel
+        return copy
+    }
```

Comment:
I think we want to be symmetrical between `withLogger` and `withBaggage`, so we also need 

```
public func withLogger(_ logger: Logger) -> DefaultContext {
         var copy = self
         copy.logger = logger
         return copy
     }
```

wdyt?

Reply:
Yeah I'll make sure the APIs are symmetric, I'll add those.

Reply:
this is still missing?

Reply:
Sorry, added 👍